dask/array/tests/test_routines.py

```diff
@@ -2526,11 +2527,24 @@ def test_tril_triu_non_square_arrays():
     [(3, 0, 3, "auto"), (3, 1, 3, "auto"), (3, -1, 3, "auto"), (5, 0, 5, 1)],
 )
 def test_tril_triu_indices(n, k, m, chunks):
-    assert_eq(
-        da.tril_indices(n=n, k=k, m=m, chunks=chunks)[0].compute(),
-        np.tril_indices(n=n, k=k, m=m)[0],
-    )
-    assert_eq(
-        da.triu_indices(n=n, k=k, m=m, chunks=chunks)[0].compute(),
-        np.triu_indices(n=n, k=k, m=m)[0],
-    )
+    actual = da.tril_indices(n=n, k=k, m=m, chunks=chunks)[0]
+    expected = np.tril_indices(n=n, k=k, m=m)[0]
+
+    if sys.platform == "win32":
+        assert_eq(
+            actual.astype(expected.dtype),
```

Comment:
Why does this difference pop up on win32 platforms?

Reply:
I am not 100% sure where, but somewhere in numpy or dask an int32 is being coerced to an int64. So this isn't specific to windows necessarily. It's just where it crops up. I was digging into it, but then I thought it might not actually be too important to people since this is not a new outcome, it is just now we are exposing it in tests.

Reply:
I believe "int" means "int32" on win32 only, "int64" everywhere else, and this is defined right in the C compiler level

Reply:
I believe "int" means "int32" on win32 only, "int64" everywhere else, and this is defined right in the C compiler level

Reply:
I believe "int" means "int32" on win32 only, "int64" everywhere else, and this is defined right in the C compiler level

Reply:
Yeah I had trouble finding where "int64" was being explicitly used.

Reply:
Thanks for the explanation @martindurant